recipes/llvm/all/conanfile.py

```diff
@@ -0,0 +1,791 @@
+            'llvm_link_llvm_dylib': False,
+            'llvm_dylib_components': 'all',
+            'fPIC': True,
+            'targets': 'all',
+            'exceptions': True,  # llvm 14 default off
+            'rtti': True,  # llvm 14 default off
+            'threads': True,
+            'lto': 'Off',
+            'static_stdlib': False,
+            'unwind_tables': True,
+            'expensive_checks': False,
+            'use_perf': False,
+            'llvm_use_sanitizer': '',
+            'with_z3': False,
+            'with_ffi': False,
+            'with_zlib': True,
+            # TODO curl 7.76.0 check is failing in config-ix.cmake
+            'with_curl': False,
+            'with_zstd': True,
+            'with_httplib': False,
+            # TODO default True, issues with liblldb.so and conan: Findlibxml2.cmake not provided
+            'with_xml2': False,
+            'keep_binaries_regex': '^$',
+
+            # options removed in package id
+            # XXX Should these files used by conan at all?
+            'use_llvm_cmake_files': False,
+            # creating job pools with current free memory
+            'ram_per_compile_job': '2000',
+            'ram_per_link_job': '14000',
+            'conan_center_index_limits': True,
+            'enable_unsafe_mode': False,
+        }
+    }
+
+    exports_sources = 'patches/**/*', 'conandata.yml'
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        apply_conandata_patches(self)
+
+    def is_windows(self):
+        return self.settings.os == "Windows"
+
+    def is_macos(self):
+        return self.settings.os == "MacOS"
+
+    def is_linux(self):
+        return self.settings.os == "Linux"
+
+    # checking options before requirements are build
+    def configure(self):
+        if self.is_windows():
+            self.options.rm_safe("fPIC")  # FPIC MANAGEMENT (KB-H007)
+
+        if self.options.shared:
+            self.options.rm_safe("fPIC")  # FPIC MANAGEMENT (KB-H007)
+            self.output.warning(
+                "BUILD_SHARED_LIBS is only recommended for use by LLVM developers. If you want to build LLVM as a shared library, you should use the LLVM_BUILD_LLVM_DYLIB option.")
+
+        release = Version(self.version).major
+        unsupported_options = []
+        if release < 14:
+            unsupported_options.append('with_curl')
+        if release < 15:
+            unsupported_options.append('with_zstd')
+            unsupported_options.append('with_httplib')
+        # TODO fix with_zstd and llvm 17
+        if release >= 17:
+            self.output.warning(f"with_zstd will fail in llvm 17 with conan 2")
+            unsupported_options.append('with_zstd')
+        for opt in unsupported_options:
+            self.output.warning(f"{opt} is unsupported in llvm {release}")
+            self.options.rm_safe(opt)
```

Comment:
It's probably better to delete options based off of the version and any `settings` in the `config_options` method instead. This way, the options won't be available when consumers try to set them in the first place. You'll want to put that above the `configure` method to reflect the evaluation order.

```python
    def config_options(self):
        if self.settings.os == "Windows":
            self.options.rm_safe("fPIC")  # FPIC MANAGEMENT (KB-H007)
        release = Version(self.version).major
        unsupported_options = []
        if release < 14:
            unsupported_options.append('with_curl')
        if release < 15:
            unsupported_options.append('with_zstd')
            unsupported_options.append('with_httplib')
        # TODO fix with_zstd and llvm 17
        if release >= 17:
            unsupported_options.append('with_zstd')
        for opt in unsupported_options:
            self.options.rm_safe(opt)
```